autoload/vader.vim

```diff
@@ -136,18 +136,43 @@ function! vader#run(bang, ...) range
   endtry
 endfunction
 
-function! s:print_stderr(output)
+function! vader#print_stderr(output) abort
   let lines = split(a:output, '\n')
   if !empty($VADER_OUTPUT_FILE)
     call writefile(lines, $VADER_OUTPUT_FILE, 'a')
   else
-    let tmp = tempname()
-    call writefile(lines, tmp)
-    execute printf('silent !%s %s 1>&2', s:cat, tmp)
-    call delete(tmp)
+    if !exists('s:tmpfile')
+      let s:tmpfile = tempname()
+    endif
+    call writefile(lines, s:tmpfile)
+    execute printf('silent !%s %s 1>&2', s:cat, s:tmpfile)
   endif
 endfunction
 
+" Overwrite vader#print_stderr with specialized version for Neovim.
+if has('nvim')
+  if !empty($VADER_ECHO_MESSAGES)
+    if $VADER_ECHO_MESSAGES ==# 'stdout'
+      let s:nvim_channel = stdioopen({})
+      function! vader#print_stderr(output) abort
+        call chansend(s:nvim_channel, a:output)
+      endfunction
+    else
+      function! vader#print_stderr(output) abort
+        call chansend(v:stderr, a:output)
+      endfunction
+    endif
+  elseif exists('*chansend') && exists('v:stderr')
+    function! vader#print_stderr(output) abort
+      call chansend(v:stderr, a:output)
+    endfunction
+  elseif exists('*nvim_list_uis') && len(nvim_list_uis()) == 0  " --headless
+    function! vader#print_stderr(output) abort
+      echon a:output
+    endfunction
+  endif
+endif
+
```

Comment:
@justinmk
Does this make sense?  (came up with this via https://github.com/neovim/neovim/issues/8846)

Reply:
yes but I wouldn't bother supporting older nvim versions without `v:stderr` (and you only need to check `exists('v:stderr')`.